### PR TITLE
I merged everything to v14 + remade the whole code and the way it works :trollface: 

### PR DESCRIPTION
**Pull request by emyfops**
Just alot of code kek

### CODE REVIEW

**Comment by kibibites** [comment]
why did you put everything into 1 commit?????

**Comment by kibibites** [comment]
im going to review it when i get home

**Comment by emyfops** [comment]
> why did you put everything into 1 commit?????

because

**Review by kibibites — COMMENTED** [review]
why was editorconfig and eslint config file removed? These file are used to format files and enfoce code style...

File: src/index.js [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+const { GatewayIntentBits } = require('discord-api-types/v10')
+const client = new Client({ intents: [
+	GatewayIntentBits.Guilds, GatewayIntentBits.GuildMembers, GatewayIntentBits.GuildInvites, GatewayIntentBits.GuildMessages, GatewayIntentBits.GuildVoiceStates, GatewayIntentBits.MessageContent
+], partials: [
+	"MESSAGE",
+	"CHANNEL",
+	"GUILD_MEMBER"
+]});
+const fs = require("fs")
+require('dotenv').config()
```
Comment (kibibites) [inline src/index.js]: what is this line supposed to do

File: package.json [hunk 2]
```diff
@@ -3,19 +3,25 @@
   "version": "1.0.0",
   "description": "",
   "main": "src/index.js",
-  "type": "module",
```
Comment (kibibites) [inline package.json:6]: you are gonna use commonjs? ugh fine

File: src/commands/capes.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    type: 3,
+                    required: true,
+                    choices: [
+                        {
+                            name: "Contributor",
+                            name: "Contributor cape",
+                            value: ""
+                        },
+                    ]
+                }*/ //anal sex
```
Comment (kibibites) [inline src/commands/capes.js]: what????
please delete

File: src/commands/capes.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                case "push": {
+                    const pushResult = await capeUtils.push();
+                    if (!pushResult) { await capeUtils.pull(); await capeUtils.push(); }
+                    return interaction.reply({ embeds: [embedUtils.success('Pushed to remote.')] })
+                }
+                case "add": {
+                    const minecraft_username = await interaction.options.getString("minecraft_username")
+                    const user = await interaction.options.getInteger("user_id")
+                    try {
+                        const _ = await axios.get(`https://discordapp.com/api/v9/users/${user}`, {
```
Comment (kibibites) [inline src/commands/capes.js]: avoid doing request here, move this into a utils file

File: src/commands/capes.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                        return await interaction.reply({ embeds: [embedUtils.error("Invalid user")] })
+                    }
+                    const minecraft_uuid = await uuidUtils.usernameToUUID(minecraft_username);
+                    console.log(minecraft_uuid)
+                    if (!minecraft_uuid) return await interaction.reply({ embeds: [embedUtils.error("Invalid username or nonexistent player")] })
+                    const addResult = capeUtils.add(user, minecraft_uuid);
+                    if (!addResult) return interaction.reply({ embeds: [embedUtils.error('No local data found. Please pull first.')] })
+                    return interaction.reply({ embeds: [embedUtils.success(`Added <@${user}>, Info: \`\`\`Username: ${minecraft_username}\nUUID: ${minecraft_uuid}\`\`\``)] })
+                }
+                default: return interaction.reply({ embeds: [embedUtils.error('Why you choose nothing :skull:')] })
```
Comment (kibibites) [inline src/commands/capes.js]: https://primer.style/design/foundations/content

File: src/commands/lookup.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+const { EmbedBuilder, ActionRowBuilder, ButtonBuilder } = require('discord.js')
+const embedUtils = require("../utils/embed.js")
+const axios = require("axios").default
+const timestampToDate = require('timestamp-to-date');
+
+ module.exports = {
+   name: "lookup",
+   description: "dLookup",
```
Comment (kibibites) [inline src/commands/lookup.js]: make a meaningful description

File: src/commands/lookup.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+   run: async(interaction) => {
+    for(let index of interaction.options.data){
+      switch(index.name){
+        case "name": {
+          const username = interaction.options.getString("username")
+          const _ = (await axios.get(`https://api.mojang.com/users/profiles/minecraft/${username}`)).data
+          if(!_.name || !_.id) return interaction.reply({ embeds: [embedUtils.error('No user found.')] })
+          const names = (await axios.get(`https://api.mojang.com/user/profiles/${_.id}/names`)).data
+          const textures = JSON.parse(Buffer.from((await axios.get(`https://sessionserver.mojang.com/session/minecraft/profile/${_.id}`)).data?.properties[0]?.value, "base64").toString())
```
Comment (kibibites) [inline src/commands/lookup.js]: as i said above, move all of these requests to a seperate utils file

File: src/commands/plugins.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+const size = require('../utils/bytes')
+
+module.exports = {
+    name: "plugins",
+    description: "Check plugin list",
+    timeout: 300000,
+
+    run: async (interaction) => {
+        let pluginList = []
+        let plugins = (await axios.get("https://api.github.com/users/lambda-plugins/repos", {
```
Comment (kibibites) [inline src/commands/plugins.js]: ^

File: src/commands/plugins.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            pluginList.push({
+                name: plugin?.name,
+                stargazers: plugin?.stargazers_count,
+                watchers: plugin?.watchers_count,
+                issues: `[Issues](plugin?.issues_url)`,
+                url: `[Repo URL](${plugin?.html_url})`,
+                size: plugin?.size
+            })
+        })
+        let _ = new EmbedBuilder().setTitle("Plugins Informations")
```
Comment (kibibites) [inline src/commands/plugins.js]: use a meaningful variable name

File: src/events.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+const { readdirSync } = require("fs");
+require('colors');
```
Comment (kibibites) [inline src/events.js:2]: why is this just required but not assigned to a variable

File: src/utils/capes.js [hunk 11]
```diff
@@ -1,33 +1,31 @@
...
+  db.sha_push(res.data.sha)
 };
 
 const push = async () => {
-  if (!fs.existsSync('../capes.json') || !fs.existsSync('../capes.json.shasum')) {
-    return 'not found';
-  }
-  octokit.request('PUT /repos/{owner}/{repo}/contents/{path}', {
+  let _ = await db.get("capes")
+  console.log(_)
```
Comment (kibibites) [inline src/utils/capes.js]: ????? too lazy to think up a variable name?

File: package.json [hunk 12]
```diff
@@ -3,19 +3,25 @@
...
-    "discord.js": "^13.7.0",
+    "better-sqlite3": "^7.5.3",
+    "discord.js": "^14.0.0-dev.1653480262-68d5169",
+    "fs": "^0.0.1-security",
     "graceful-fs": "^4.2.10",
+    "humanize-duration": "^3.27.2",
     "ms": "^2.1.3",
     "node-fetch": "^3.2.4",
-    "path": "^0.12.7"
+    "octokit": "^1.7.1",
```
Comment (kibibites) [inline package.json]: we already have octokit line 12

File: src/commands/plugins.js (hunk 9 above)
Comment (ToxicAven) [inline src/commands/plugins.js]: Bro what why `_` variables

File: src/utils/quickdb.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    db.set("sha", data)
+}
+
+const get = async(key = String) => {
+    init();
+    let _ = await db.get(key)
+    return _ != null ? _ : false
+}
+
+const UwU = {
```
Comment (kibibites) [inline src/utils/quickdb.js]: LMFAO????

File: .editorconfig [hunk 14]
```diff
@@ -1,5 +0,0 @@
-root = true
-
-[*.{js,cjs}]
```
Comment (ToxicAven) [inline .editorconfig:3]: This should be kept because formatting

File: src/utils/uuid.js [hunk 15]
```diff
@@ -1,14 +1,21 @@
...
   const part2 = uuid.slice(8, 12);
   const part3 = uuid.slice(12, 16);
   const part4 = uuid.slice(16, 20);
   const part5 = uuid.slice(20, 32);
-  return `${part1}-${part2}-${part3}-${part4}-${part5}`;
+  return `${part1}-${part2}-${part3}-${part4}-${part5}`
 };
+const usernameToUUID = async(username) => {
+  const axios = require("axios").default
+  const uuid = await axios.get(`https://api.mojang.com/users/profiles/minecraft/${username}`).catch(e => console.log(e.data))
```
Comment (kibibites) [inline src/utils/uuid.js]: make this into a seperate utils file

**Review by kibibites — CHANGES_REQUESTED** [review]

File: package.json [hunk 16]
```diff
@@ -3,19 +3,25 @@
...
   "author": "i-spin",
   "license": "Unlicense",
   "dependencies": {
     "@octokit/core": "^3.6.0",
-    "discord.js": "^13.7.0",
+    "better-sqlite3": "^7.5.3",
+    "discord.js": "^14.0.0-dev.1653480262-68d5169",
+    "fs": "^0.0.1-security",
     "graceful-fs": "^4.2.10",
+    "humanize-duration": "^3.27.2",
```
Comment (kibibites) [inline package.json:21]: just use ms

File: package.json [hunk 17]
```diff
@@ -3,19 +3,25 @@
...
   "scripts": {
     "dev": "cd src && node index.js"
   },
   "author": "i-spin",
   "license": "Unlicense",
   "dependencies": {
     "@octokit/core": "^3.6.0",
-    "discord.js": "^13.7.0",
+    "better-sqlite3": "^7.5.3",
+    "discord.js": "^14.0.0-dev.1653480262-68d5169",
```
Comment (ToxicAven) [inline package.json:18]: Why?

File: src/slash.js [hunk 18]
```diff
@@ -0,0 +1,32 @@
+const { REST } = require('@discordjs/rest');
+const { Routes } = require('discord-api-types/v10');
+const { readdirSync } = require('fs');
```
Comment (ToxicAven) [inline src/slash.js]: readDirSync

File: src/commands/capes.js (hunk 3 above)
Comment (emyfops) [reply src/commands/capes.js]: kek

File: src/commands/lookup.js (hunk 6 above)
Comment (emyfops) [reply src/commands/lookup.js]: I forgot to edit

File: src/commands/plugins.js (hunk 9 above)
Comment (emyfops) [reply src/commands/plugins.js]: "_" is meant for temp variable with no real uses

File: src/utils/quickdb.js [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+const safecheck = require('./checkjson')
+const sha1 = require('sha1')
+
+const init = async () => {
+    if(!db.get("capes")){ db.set("capes", [])}
+}
+
+const push = async(key, ...data) => {
+    init();
+    let __ = []
```
Comment (ToxicAven) [inline src/utils/quickdb.js]: how hard is it to come up with readable variable names

File: src/utils/capes.js [hunk 20]
```diff
@@ -62,8 +59,16 @@ const add = async (discordId, uuid) => {
       },
     ],
     is_premium: true,
-  });
-  fs.writeFileSync('../capes.json', JSON.stringify(data, null, 2));
+  }
+  try {
+    _.push(template)
```
Comment (ToxicAven) [inline src/utils/capes.js]: underscore variable again

File: src/commands/lookup.js (hunk 6 above)
Comment (kibibites) [reply src/commands/lookup.js]: i forgor :skull:

File: src/utils/capes.js [hunk 21]
```diff
@@ -1,33 +1,31 @@
...
+  db.push("capes", Buffer.from(res.data.content, 'base64').toString())
+  db.sha_push(res.data.sha)
 };
 
 const push = async () => {
-  if (!fs.existsSync('../capes.json') || !fs.existsSync('../capes.json.shasum')) {
-    return 'not found';
-  }
-  octokit.request('PUT /repos/{owner}/{repo}/contents/{path}', {
+  let _ = await db.get("capes")
```
Comment (ToxicAven) [inline src/utils/capes.js]: Underscores :/

File: src/commands/capes.js (hunk 4 above)
Comment (emyfops) [reply src/commands/capes.js]: It's useless to create a file for 4 lines of code

File: src/commands/ping.js [hunk 22]
```diff
@@ -0,0 +1,14 @@
+const embedUtils = require('../utils/embed')
+
+module.exports = {
+	name: 'stats',
```
Comment (ToxicAven) [inline src/commands/ping.js]: is this skidded?

File: src/commands/lookup.js [hunk 23]
```diff
@@ -0,0 +1,68 @@
...
+
+          const embed = new EmbedBuilder()
+          .setTitle(`${username}'s UUID`)
+          .addFields([{ name: "UUID:", value: _.id }])
+          .setImage(`https://crafatar.com/renders/body/${_.id}?overlay`)
+          .setColor("Blue");
+          try {
+            
+          names.forEach((name) => {
+            let _ = timestampToDate(name.changedToAt,'yyyy-MM-dd HH:mm:ss');if(_.toLowerCase().includes('nan')) _ = "First Appeared Name"
```
Comment (ToxicAven) [inline src/commands/lookup.js]: underscore

File: src/commands/capes.js (hunk 3 above)
Comment (ToxicAven) [inline src/commands/capes.js]: bruh

**Review by ToxicAven — CHANGES_REQUESTED** [review]
This is possibly one of the weirdest PRs i've dealt with, how and why

File: src/events.js (hunk 10 above)
Comment (emyfops) [reply src/events.js:2]: Cause It's an extend to String
I can use console.log(String.green)

File: src/index.js (hunk 1 above)
Comment (emyfops) [reply src/index.js]: Read the Discord v14 Documentation

File: src/utils/quickdb.js (hunk 13 above)
Comment (emyfops) [reply src/utils/quickdb.js]: Easy database interaction, nothing much to say

File: src/utils/uuid.js (hunk 15 above)
Comment (emyfops) [reply src/utils/uuid.js]: It is already in a different file

File: src/commands/capes.js (hunk 5 above)
Comment (emyfops) [reply src/commands/capes.js]: You can't even access the default one
`for (let index of interaction.options.data) {}`

File: src/index.js (hunk 1 above)
Comment (emyfops) [reply src/index.js]: My bad, dotenv is an implementation in process

File: src/commands/capes.js (hunk 4 above)
Comment (emyfops) [reply src/commands/capes.js]: Fixed

File: src/commands/capes.js (hunk 5 above)
Comment (emyfops) [reply src/commands/capes.js]: Fixed

File: src/commands/lookup.js (hunk 6 above)
Comment (emyfops) [reply src/commands/lookup.js]: Fixed

File: src/commands/ping.js (hunk 22 above)
Comment (emyfops) [reply src/commands/ping.js]: I want to add more stats, not only ping

File: src/slash.js (hunk 18 above)
Comment (emyfops) [reply src/slash.js]: I don't need the whole fs

File: .editorconfig (hunk 14 above)
Comment (emyfops) [reply .editorconfig:3]: I don't even know what that is

**Review by kibibites — APPROVED** [review]
just need to fix the dyn requires that are everywhere

**Pull request merged**